### PR TITLE
Verify the test output

### PR DESCRIPTION
**Pull request by Link1515**
#33 

This PR adds tests to ensure the correct behavior of the logger. The tests cover a variety of scenarios, including:

- Log Chain
- Utility Methods
- Level Log
- Steam
- Environment Variable

The tests were written using [Jest](https://jestjs.io/) and are located in the `test/jest` directory.

### CODE REVIEW

File: test/jest/environmentVariable.spec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    getLevelLogMessage, 
+    getLevelLogTimestamp,
+    createLevelLogMessage
+} = require('./utils/levelLogUtils');
+
+logger._customizedConsole.error = () => {}
+
+describe('Environment Variable LOGGER', () => {
+    test('LOGGER=info, debug and success level will not show', () => {
+        logger.error('error show');
```
Comment (tigercosmos) [inline test/jest/environmentVariable.spec.js:13]: `error`, and `warn` seems unnecessary here.

File: test/jest/environmentVariable.spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        logger.warn('warn show');
+        logger.info('info show');
+        logger.debug('debug will not show');
+        logger.success('success will not show');
+
+        expect(
+            Date.now() - getLevelLogTimestamp(logger.lastCommand),
+            'Log time error exceeds 5 seconds'
+        ).toBeLessThan(5 * 1000);
```
Comment (tigercosmos) [inline test/jest/environmentVariable.spec.js:22]: why 5 seconds? should be less than 100ms, or even not setting the timeout?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
+/**
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
+ */
+
+/** @type {import('jest').Config} */
+const config = {
```
Comment (tigercosmos) [inline jest.config.js:7]: do you have any customized settings? if not, maybe no need this file?

File: test/jest/levelLog.spec.js [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+    test('set level to info', () => {
+        logger.setLevel('info');
+        expect(logger.level).toBe('info');
+
+        logger.error('error show');
+        expect(
+            Date.now() - getLevelLogTimestamp(logger.lastCommand),
+            'Log time error exceeds 5 seconds'
+        ).toBeLessThan(5 * 1000);
```
Comment (tigercosmos) [inline test/jest/levelLog.spec.js:101]: I wonder if we can remove all timeouts?

**Comment by tigercosmos** [comment]
@Link1515  Thanks for your great work. Overall it looks pretty good to me.  Just some comments above, could you take a look?

File: test/jest/levelLog.spec.js (hunk 4 above)
Comment (Link1515) [reply test/jest/levelLog.spec.js:101]: Sure!

File: jest.config.js (hunk 3 above)
Comment (Link1515) [reply jest.config.js:7]: In the jest.config.js, I do the following setting:

- Using jest-expect-message package to write custom message

```javascript
{
  setupFilesAfterEnv: ['jest-expect-message'],
}
```

- Specify the file which should be executed by jest (file with *.spec.js)
  - If not specified, test/test.js will also be executed by default

```javascript
{
  testMatch: [
    "**/test/**/*.spec.js"
  ],
}
```

**Comment by tigercosmos** [comment]
@Link1515  thanks!

**Pull request merged**